backend/core/src/auth/services/user.service.ts

```diff
@@ -300,9 +312,41 @@ export class UserService {
     }
   }
 
-  public isUserConfirmationTokenValid(dto: ConfirmDto) {
+  public async recreatePartnerUnconfirmedUserTokens(params: ConfirmationListQueryParams) {
+    const users = await this.findAll({ confirmedAt: null })
+    const userTokens: { email: string; confirmationUrl: string }[] = []
+
+    const recreateToken = async (user: User) => {
+      user.confirmationToken = UserService.createConfirmationToken(user.id, user.email)
+      try {
+        await this.userRepository.save(user)
+        const confirmationUrl = UserService.getPartnersConfirmationUrl(params.appUrl, user)
+        userTokens.push({ email: user.email, confirmationUrl })
+        if (params.sendEmail) {
+          await this.emailService.invite(user, params.appUrl, confirmationUrl)
+        }
+        return user
+      } catch (err) {
+        throw new HttpException(USER_ERRORS.ERROR_SAVING.message, USER_ERRORS.ERROR_SAVING.status)
+      }
+    }
+
+    for (const user of users) {
+      await recreateToken(user)
```

Comment:
thinking about this a bit more, I think we may want to change this to leverage https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled so instead of waiting for each record to be saved then emailed consecutively we can do them all async and just wait for all of them to finish up 

it could save us time and make this process faster when being ran against a larger set of users

Reply:
@YazeedLoonat do you think the performance is a blocker or would cause fxnality issues? just wondering since i think we will hopefully run this once and close this pr

Reply:
@seanmalbert @emilyjablonski so I ran some benchmarking here and the findings are as follows:

against dev each run of the `recreateToken` function took from 800-880ms and total time to run against dev was  15.52 seconds. There were 6 users that got updated

so I think we are likely to run into the 2 minute timer for the approx 120 users that will need to be updated 

like sean mentioned we can either up the max request time, or we can switch this to a promise.all/promise.allSettled

Reply:
@YazeedLoonat which do you recommend / would you be able to make that change?